packages/service-library/src/servicelib/redis/_client.py

```diff
@@ -77,15 +81,22 @@ async def _periodic_check_health() -> None:
         )
 
     async def shutdown(self) -> None:
-        if self._health_check_task:
-            assert self._health_check_task_started_event  # nosec
-            await self._health_check_task_started_event.wait()  # NOTE: wait for the health check task to have started once before we can cancel it
-            await cancel_wait_task(self._health_check_task)
-
-        await self._client.aclose(close_connection_pool=True)
+        with log_context(
+            _logger, level=logging.DEBUG, msg=f"Shutdown RedisClientSDK {self}"
+        ):
+            if self._health_check_task:
+                assert self._health_check_task_started_event  # nosec
+                # NOTE: wait for the health check task to have started once before we can cancel it
+                await self._health_check_task_started_event.wait()
+                await cancel_wait_task(
+                    self._health_check_task, max_delay=HEALTHCHECK_TASK_TIMEOUT_S
```

Comment:
Q: was this really preventing the test from running or that is a side issue? I find this very strange. could this be something to do with that sync/async issue?

Reply:
Not 100% sure but tests starting passing after these changes. As I mention in the description above my guess are the interaction of tasks and the services around